sunpy/net/hek/tests/test_hek.py

```diff
@@ -228,3 +276,79 @@ def test_missing_times():
     assert isinstance(results["event_peaktime"][0], np.ma.core.MaskedConstant)
     assert results["event_peaktime"][6].isot == "2024-05-10T16:08:00.000"
+
+@pytest.mark.remote_data
+def test_merging_event_coords():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    client = hek.HEKClient()
+    result = client.search(attrs.Time(tstart,tend), attrs.hek.EventType('CH'))
+
+    coord1 = -2.91584*u.arcsec
+    coord2 = 940.667*u.arcsec
+    time='2011-08-09 06:00:08.000'
+    frame='helioprojective'
+    event_coord = SkyCoord(coord1, coord2, obstime=time, frame=frame)
+
+    assert result['event_coord'][0] == event_coord
+
+@pytest.mark.remote_data
+def test_obs_meanwavel():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    client = hek.HEKClient()
+    result = client.search(attrs.Time(tstart,tend), attrs.hek.EventType('CH'))
+
+    value = 193.0*u.angstrom
+    assert result['obs_meanwavel'][0] == value
+
+@pytest.mark.remote_data
+def test_parse_times():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    client = hek.HEKClient()
+    result = client.search(attrs.Time(tstart,tend), attrs.hek.EventType('CH'))
+
+    for idx in range(len(result)):
+        assert isinstance(result['event_starttime'][idx], Time)
+        assert isinstance(result['event_endtime'][idx], Time)
+
+@pytest.mark.remote_data
+def test_ssw_latest_events_flares():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.FRM.Name == 'SSW Latest Events')
+    assert len(result[0]) == 2
+
+@pytest.mark.remote_data
+def test_not_ssw_latest_events_flares():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.FRM.Name != 'SSW Latest Events')
+    assert len(result[0]) == 19
+
+@pytest.mark.remote_data
+def test_flares_peak_flux():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.FL.PeakFlux > 4000.0)
+    assert len(result[0]) == 1
+
+@pytest.mark.remote_data
+def test_flares_peak_flux_and_position():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.Event.Coord1 > 800, attrs.hek.FL.PeakFlux > 1000)
+    assert len(result[0]) == 7
+
+@pytest.mark.remote_data
+def test_flares_python_logical_ops():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), (attrs.hek.Event.Coord1 > 50) and (attrs.hek.FL.PeakFlux > 1000))
+    assert len(result[0]) == 7
```

Comment:
This applies to all of these style of tests, if we are filtering by coord or flux or any other value. 

We should check the from the return as well and not just the length.